File: 46/source.java
public class e46 {
	
	static boolean[] primes  = new boolean[100000001];
	static final int NUM = 1000000;
	
	private static void fillPrimes(int num){
		for(int i=0; i < num;i++){
		primes[i] = (!(i == 2 || i%2 == 0 || i == 0 || i == 1));
		}
		for(int i = 3; i <= (num/2); i = i + 2){
			if(primes[i]){
				for(int j= 2*i; j < num; j += i){
					primes[j] = false;
					}
				}
			}
		primes[2] = true;
	}
	
	private static boolean goldbach(int a){
		int result;
		result = 0;
		for(int i = 2; i < a; i++){
			if(primes[i]){
				for(int j = 1; j < a; j++){
					result = (int) (i + (2*(Math.pow(j, 2))));
					if(result == a){
						return true;
					}
				}
			}
		}
		return false;
	}
	
	public static void main(String[] args) {
		fillPrimes(NUM);
		int result;
		result = 0;
		for(int i = 2; i < NUM; i++){
			if(!primes[i] && i%2 != 0){
				if(!goldbach(i)){
					result = i;
					break;
				}
			}
		}
		System.out.println("Result: " + result);
	}
}

File: 47/source.py
from math import sqrt

def erathostenos(num):
    primes = { i:(not(i == 2 or i % 2 == 0 or i == 0 or i == 1)) for i in range(num+1) }
    primes[2] = True
    limit   = sqrt(num) + 1
    i       = 3
    while i <= limit:
        if primes[i]:
            j = 2 * i
            while j < num:
                primes[j] = False
                j += i
        i += 2
    return primes

# generate list with primes
primes = [i for i,j in erathostenos(100000).items() if j ]

def checkPrimeFactors(a, b, c, d):
    factors = set()
    for i in primes:
        tmp = 1
        while a%i == 0:
            a = a // i
            tmp *= i
        if tmp != 1:
            factors.add(tmp)
        if a == 1:
            break

    for i in primes:
        tmp = 1
        while b%i == 0:
            tmp *= i
            b = b // i
        if tmp != 1:
            factors.add(tmp)
        if b == 1:
            break

    for i in primes:
        tmp = 1
        while c%i == 0:
            tmp *= i
            c = c // i
        if tmp != 1:
            factors.add(tmp)
        if c == 1:
            break

    for i in primes:
        tmp = 1
        while d%i == 0:
            tmp *= i
            d = d // i
        if tmp != 1:
            factors.add(tmp)
        if d == 1:
            break

    return len(factors) == 16 # 16 because we need 4 distinct prime factors for each number, a, b, c and d

for i in range(100000, 200000):
    if checkPrimeFactors(i, i+1, i+2, i+3):
        print("Result: ", i)
        break;